internal/driver/driver.go

```diff
@@ -329,9 +365,12 @@ func (s *Driver) AddDevice(deviceName string, protocols map[string]models.Protoc
 // when a Device associated with this Device Service is updated
 func (s *Driver) UpdateDevice(deviceName string, protocols map[string]models.ProtocolProperties, adminState models.AdminState) error {
 	s.lc.Debugf("Device %s is updated", deviceName)
+
+	s.s7Clients[deviceName] = nil
```

Comment:
I don't think this is needed since you overwrite the value on line 376. Otherwise it needs locking around it.

Reply:
Yes, you're right. 👍